eras/conway/impl/src/Cardano/Ledger/Conway/CDDL.hs

```diff
@@ -0,0 +1,971 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{-# HLINT ignore "Use camelCase" #-}
+{-# HLINT ignore "Evaluate" #-}
+
+module Cardano.Ledger.Conway.CDDL where
+
+import Codec.CBOR.Cuddle.Huddle
+import Data.Function (($))
+import Data.Semigroup ((<>))
+import Data.Text qualified as T
+import Data.Word (Word64)
+import GHC.Float (Double)
+import GHC.Num (Integer)
+import GHC.Show (Show (show))
+
+default (Integer, Double)
+
+conway :: Huddle
+conway = collectFrom [block, transaction]
+
+block :: Rule
+block =
+  "block"
+    =:= arr
+      [ a header
+      , "transaction_bodies" ==> arr [0 <+ a transaction_body]
+      , "transaction_witness_sets"
+          ==> arr [0 <+ a transaction_witness_set]
+      , "auxiliary_data_set"
+          ==> mp [0 <+ asKey transaction_index ==> auxiliary_data]
+      , "invalid_transactions" ==> arr [0 <+ a transaction_index]
+      ]
+
+transaction :: Rule
+transaction =
+  "transaction"
+    =:= arr
+      [ a transaction_body
+      , a transaction_witness_set
+      , a VBool
+      , a (auxiliary_data // VNil)
+      ]
+
+transaction_index :: Rule
+transaction_index = "transaction_index" =:= VUInt `sized` (2 :: Word64)
+
+header :: Rule
+header = "header" =:= arr [a header_body, "body_signature" ==> kes_signature]
+
+header_body :: Rule
+header_body =
+  "header_body"
+    =:= arr
+      [ "block_number" ==> VUInt
+      , "slot" ==> VUInt
+      , "prev_hash" ==> hash32
+      , "issuer_vkey" ==> vkey
+      , "vrf_vkey" ==> vkey
+      , "vrf_result" ==> vrf_cert
+      , "block_body_size" ==> VUInt
```

Comment:
Are we using `VUInt` to represent all sizes?  Is this the same VUint defined like this in cuddle?
```
data Value
  = VUInt Word64
  | VNInt Word64
   ..
```

How does it work then,  can we define precise sizes for the unsigned integers as we do in the cddl file?

Reply:
This is a very important question. We do need ability to specify sizes, eg. [`block_body_size  : uint .size 4`](https://github.com/IntersectMBO/cardano-ledger/blob/f4119c9cd27e69943dcf91d22d0f9905be32fa75/eras/conway/impl/cddl-files/conway.cddl#L34)

I assume it can be achieved with something like this, right?
```suggestion
      , "block_body_size" ==> VUInt `sized` (4 :: Word)
```